Use switch over a span in a few more places

stephentoub opened the pull request:
C# has added the ability to `switch` on a `ReadOnlySpan<char>`, using cascading `SequenceEquals` calls when there are only a few cases, and using the same hash-based jump table when there are enough cases.  Use it in a few more places.

vcsjones commented on src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/WebSocketHandle.Managed.cs:
```diff
@@ -257,21 +257,23 @@ private static WebSocketDeflateOptions ParseDeflateOptions(ReadOnlySpan<char> ex
-                    {
-                        options.ServerContextTakeover = false;
-                    }
-                    else if (value.StartsWith(ClientWebSocketDeflateConstants.ClientMaxWindowBits))
```
These are `StartsWith`, not `SequenceEqual`.

stephentoub replied:
Ugh, so they are.

bartonjs replied:
And my order of System.Security.*/... to the end and then get "the file I skipped over" reveals a flaw...

stephentoub committed:
Use switch over a span in a few more places

C# has added the ability to `switch` on a `ReadOnlySpan<char>`, using cascading `SequenceEquals` calls when there are only a few cases, and using the same hash-based jump table when there are enough cases.  Use it in a few more places.
